src/main/scala/inox/solvers/unrolling/UnrollingSolver.scala

```diff
@@ -164,8 +164,15 @@ trait AbstractUnrollingSolver extends Solver { self =>
 
   def assertCnstr(expression: Expr): Unit = context.timers.solvers.assert.run(try {
     context.timers.solvers.assert.sanity.run {
-      symbols.ensureWellFormed // make sure that the current program is well-formed
-      typeCheck(expression, BooleanType()) // make sure we've asserted a boolean-typed expression
+      try {
+        symbols.ensureWellFormed // make sure that the current program is well-formed
+        typeCheck(expression, BooleanType()) // make sure we've asserted a boolean-typed expression
+      } catch {
+        case _: Throwable =>
+          context.reporter.internalError(
```

Comment:
Can't we include the original error in the exception? This will make debugging harder since we'll lose the original stack trace.

Reply:
After some more thought, I think it would make more sense to do the try-catch inside Stainless. This gives more flexibility to Inox clients about how they want to handle/display the error.